includes/sanitizers/class-amp-video-sanitizer.php

```diff
@@ -32,21 +42,46 @@ public function sanitize() {
 			foreach ( $node->childNodes as $child_node ) {
 				$new_child_node = $child_node->cloneNode( true );
 				$old_child_attributes = AMP_DOM_Utils::get_node_attributes_as_assoc_array( $new_child_node );
+
+				if ( isset( $old_child_attributes['src'] ) ) {
+					$orig_src[] = $old_child_attributes['src'];
+				}
+
 				$new_child_attributes = $this->filter_attributes( $old_child_attributes );
 
 				// Only append source tags with a valid src attribute
 				if ( ! empty( $new_child_attributes['src'] ) && 'source' === $new_child_node->tagName ) {
+					AMP_DOM_Utils::add_attributes_to_node( $new_child_node, $new_child_attributes );
 					$new_node->appendChild( $new_child_node );
 				}
 			}
 
 			// If the node has at least one valid source, replace the old node with it.
+			// If the node has no valid sources, but at least one invalid (http) one, add a fallback element.
 			// Otherwise, just remove the node.
-			//
-			// TODO: Add a fallback handler.
-			// See: https://github.com/ampproject/amphtml/issues/2261
 			if ( 0 === $new_node->childNodes->length && empty( $new_attributes['src'] ) ) {
-				$node->parentNode->removeChild( $node );
+				if ( ! empty( $orig_src ) ) {
+					$fallback_node = AMP_DOM_Utils::create_node(
+						$this->dom,
+						'blockquote',
+						array(
+							'class' => 'amp-wp-fallback amp-wp-video-fallback'
+						)
+					);
+
+					$fallback_content = $this->dom->createDocumentFragment();
+					$fallback_content->appendXML( sprintf(
+							wp_kses( __( 'Could not load <a href="%s">video</a>.', 'amp' ), array( 'a' => array( 'href' => true ) ) ),
+							esc_url( array_shift( $orig_src ) )
+						)
+					);
+
+					$fallback_node->appendChild( $fallback_content );
+
+					$node->parentNode->replaceChild( $fallback_node, $node );
+				} else {
+					$node->parentNode->removeChild( $node );
+				}
```

Comment:
This block of code (and the similar ones across the other sanitizers) could probably be abstracted to a new method?